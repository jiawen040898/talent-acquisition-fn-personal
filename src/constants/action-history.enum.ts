export const DEFAULT_CANDIDATE_USERNAME = 'candidate';
