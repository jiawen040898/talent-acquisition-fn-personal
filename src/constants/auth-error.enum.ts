export enum AuthErrorMessage {
    FAILED_GET_TOKEN = 'Error while trying to obtain access token from auth0',
}
