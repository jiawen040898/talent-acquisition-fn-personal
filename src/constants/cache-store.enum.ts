export enum CacheObject {
    FIT_SCORE_RECIPE_ID = 'FIT_SCORE_RECIPE:RECIPE_ID',
}
