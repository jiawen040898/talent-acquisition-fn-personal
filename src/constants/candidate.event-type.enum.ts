export enum CandidateEventType {
    APPLICATION_SIGNUP = 'candidate_application_signup',
    APPLICATION_APPLIED = 'candidate_application_applied',
    APPLICATION_SUBMITTED = 'candidate_application_submitted',
    APPLICATION_COMPLETED = 'candidate_application_completed',
    APPLICATION_ASSESSMENT_COMPLETED = 'candidate_application_assessment_completed',
    ASSESSMENT_STARTED = 'candidate_assessment_started',
    ASSESSMENT_RESETTED = 'candidate_assessment_resetted',
    ASSESSMENT_SUBMITTED = 'candidate_assessment_submitted',
    JOB_APPLICATION_APPLIED = 'job_application_applied',
    APPLICATION_RESUME_UPLOADED = 'candidate_application_resume_uploaded',
    APPLICATION_WITHDRAWN = 'candidate_application_withdrawn',
    APPLICATION_UPDATED = 'candidate_application_updated',
}
