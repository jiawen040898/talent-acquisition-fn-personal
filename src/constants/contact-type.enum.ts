export enum ContactType {
    FACEBOOK = 'facebook',
    LINKED_IN = 'linkedin',
    TWITTER = 'twitter',
    GOOGLE_OAUTH2 = 'google-oauth2',
    GITHUB = 'github',
    EMAIL = 'email',
    MOBILE = 'mobile',
    GOOGLE_APPS = 'google-apps',
    AUTH0 = 'auth0',
    INSTAGRAM = 'instagram',
    GOOGLE_PLUS = 'google-plus',
    WEBSITE = 'website',
    OTHERS = 'others',
}
