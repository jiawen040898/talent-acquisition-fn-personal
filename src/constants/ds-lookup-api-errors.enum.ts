export enum DSLookupAPIErrorCode {
    FAILED_TO_GET_PAIRWISE_SIMILARITY = 'failed_to_get_pairwise_similarity',
}

export enum DSLookupAPIErrorType {
    FAILED_TO_GET_PAIRWISE_SIMILARITY = 'FailedToGetPairwiseSimilarity',
}

export enum DSLookupAPIErrorMessage {
    FAILED_TO_GET_PAIRWISE_SIMILARITY = 'Failed to get pairwise similarity',
}
