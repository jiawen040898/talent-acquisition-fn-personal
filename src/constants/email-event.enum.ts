export enum EmailEventType {
    SEND_EMAIL_TRIGGERED = 'send_email_triggered',
}
