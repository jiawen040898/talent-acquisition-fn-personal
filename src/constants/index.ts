export * from './action-history.enum';
export * from './auth-error.enum';
export * from './cache-store.enum';
export * from './candidate.event-type.enum';
export * from './channel.enum';
export * from './contact-type.enum';
export * from './ds-lookup-api-errors.enum';
export * from './email-event.enum';
export * from './email-type.enum';
export * from './error-msg.enum';
export * from './errors';
export * from './feature-toggle-name.enum';
export * from './fit-score-recipe.enum';
export * from './identity-event.enum';
export * from './job.enum';
export * from './job-application-score-type.enum';
export * from './job-application-skill.enum';
export * from './job-application-status.enum';
export * from './person-score.enum';
export * from './person-score-event.enum';
export * from './psychology-api-errors.enum';
export * from './questionnaire-framework.enum';
export * from './s3-errors.enum';
export * from './screening-answer-criteria.enum';
export * from './skill-extract-api-errors.enum';
export * from './talent-acquisition-event.enum';
export * from './talent-management-event.enum';
export * from './timeline.enum';
export * from './unleash-errors.enum';
export * from './user.enum';
