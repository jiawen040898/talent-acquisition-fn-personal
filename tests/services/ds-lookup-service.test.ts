import { DSLookupAPIErrorMessage } from '@pulsifi/constants';
import {
    DSLookupService,
    FailedToGetPairwiseSimilarityFromDSLookupApi,
} from '@pulsifi/services';
import axios from 'axios';

import { testData } from './fixtures/test-data';

describe('DSLookupService', () => {
    const mockHttpServicePost = jest.spyOn(axios, 'post');

    describe('getPairwiseSimilarity', () => {
        beforeEach(() => {
            mockHttpServicePost.mockClear();
        });

        it('should get pairwise similarity for hard skill', async () => {
            // Arrange
            mockHttpServicePost.mockResolvedValue(
                testData.mockGetPairwiseSimilarityHardSkill,
            );
            const payload = {
                job_required_skills: ['communication', 'business management'],
                job_application_skills: ['communication'],
            };

            // Act
            const actual = await DSLookupService.getPairwiseSimilarity(
                payload.job_required_skills,
                payload.job_application_skills,
            );

            // Assert
            expect(actual).toMatchSnapshot();
        });

        it('should get pairwise similarity for work experience', async () => {
            // Arrange
            mockHttpServicePost.mockResolvedValue(
                testData.mockGetPairwiseSimilarityWorkExperience,
            );
            const payload = {
                job_title: ['software engineer', 'sales manager '],
                job_application_roles: ['software engineer'],
            };

            // Act
            const actual = await DSLookupService.getPairwiseSimilarity(
                payload.job_title,
                payload.job_application_roles,
            );

            // Assert
            expect(actual).toMatchSnapshot();
        });

        it('should throw error when get pairwise similarity failed', async () => {
            // Arrange
            mockHttpServicePost.mockRejectedValue(new Error());
            const payload = {
                job_title: ['software engineer', 'sales manager'],
                job_application_roles: ['software engineer'],
            };

            // Act
            const action = () =>
                DSLookupService.getPairwiseSimilarity(
                    payload.job_title,
                    payload.job_application_roles,
                );

            // Assert
            await expect(action).rejects.toThrow(
                DSLookupAPIErrorMessage.FAILED_TO_GET_PAIRWISE_SIMILARITY,
            );

            await expect(action).rejects.toBeInstanceOf(
                FailedToGetPairwiseSimilarityFromDSLookupApi,
            );
        });
    });
});
