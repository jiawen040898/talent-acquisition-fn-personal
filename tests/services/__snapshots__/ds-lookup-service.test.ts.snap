// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSLookupService getPairwiseSimilarity should get pairwise similarity for hard skill 1`] = `
{
  "pairwise_score": {
    "Liaising": {
      "Career": 0.26674896478652954,
      "Communication": 0.5105934739112854,
      "Design": 0.3413463830947876,
    },
  },
  "score": 0.521033439371321,
}
`;

exports[`DSLookupService getPairwiseSimilarity should get pairwise similarity for work experience 1`] = `
{
  "pairwise_result": [
    {
      "matches": [
        {
          "job_title": "Engineer",
          "score": 0.33818328380584717,
        },
      ],
      "previous_employment": "BUSINESS MANAGER",
    },
  ],
}
`;
